widget/view.controller.js

```diff
@@ -85,208 +70,146 @@
-              if (linkIndex !== -1) {
-                links[linkIndex].value = links[linkIndex].value + link.total;
-              } else {
-                let linkData = {
-                  "source": nodesMap['s_' + link.source],
-                  "target": nodesMap['t_' + link.target],
-                  "value": link.total,
-                  "id": id
-                };
-                if (nodeColorMap[link.target]) {
-                  linkData.color = nodeColorMap[link.target];
-                }
-                links.push(linkData);
-              }
-            }
-            if (link.target && link.relation) {
-              let linkData = {
-                "source": nodesMap['t_' + link.target],
-                "target": nodesMap['r_' + link.relation],
-                "value": link.total,
-                "id": id
-              };
-              if (nodeColorMap[link.target]) {
-                linkData.color = nodeColorMap[link.relation];
-              }
-              links.push(linkData);
+        }
+      });
+      nodes.forEach(node => {
+        if (nodeColorMap[node.name]) {
+          node.color = nodeColorMap[node.name];
+        }
+      });
+      //fetchJSONData();
+      // Append input JSON data for Source/External Json nodes
+      // ToDo: Add API call to fetch JSON data in global variable, if API fail, then accept data from JSON field
+      
+      // create nodes and links data as per the dynamic series generated data
+      dataToPlot.forEach((link,index) => {
+        for (let c = 0; c < seriesNameArray.length; c++) {
+          if (link[seriesNameArray[c]] && link[seriesNameArray[c+1]]) {
+            let linkData = {
+              "source": _.filter(nodesMap, { 'name': link[seriesNameArray[c]] + '_' + c})[0]['idIndex'],
+              "target": _.filter(nodesMap, { 'name': link[seriesNameArray[c+1]] + '_' + (c+1) })[0]['idIndex'],
+              "value": link.total,
+              "id": index
+            };
+            if (nodeColorMap[link[seriesNameArray[c]]]) {
+              linkData.color = nodeColorMap[link[seriesNameArray[c+1]]];
             }
-            idIndex++;
-          });
-
-          chartData.nodes = nodes;
-          chartData.links = links;
-
-          renderSankeyChart();
-        } else {
-          renderNoRecordMessage();
+            links.push(linkData);
+          }
         }
-      }, function (error) {
-        $scope.processing = false;
-        renderNoRecordMessage();
-      }).finally(function () {
-        $scope.processing = false;
       });
+
+      if(nodes.length > 0 && links.length>0){
+        chartData.nodes = nodes;
+        chartData.links = links;
+        renderSankeyChart();
+      }else{
+        errorMessage = $scope.viewWidgetVars.MESSAGE_NO_NODES_LINKS;
+        renderNoRecordMessage('Nodes and Links not created by the given data')
```

Comment:
Here the message is hardcoded. Need to pass localized error message here.

Reply:
parameter passing removed from function